### ISSUE TITLE
Resume loads the original custom `pretrained` weights instead of the resume checkpoint

### ISSUE THREAD

**Issue by geiyiren** [ultralytics/ultralytics#25812]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Train

### Issue title

`resume=True` silently overwrites resume checkpoint weights when `pretrained` is a path string

### Summary

When a training run is started with a custom pretrained checkpoint path, for example
`pretrained="path/to/pretrained.pt"`, that string is saved in the training checkpoint arguments. On resume,
`BaseTrainer.check_resume()` restores the saved `pretrained` string. `BaseTrainer.setup_model()` then loads the resume
checkpoint first, but immediately replaces its model weights with the original `pretrained` checkpoint.

The optimizer, scaler, EMA, epoch, and best fitness are still restored from the resume checkpoint. The resulting state
is therefore inconsistent:

| State | Loaded from |
| --- | --- |
| Training model weights | Original custom `pretrained.pt` |
| Optimizer state | Resume `last.pt` |
| AMP scaler state | Resume `last.pt` |
| EMA state | Resume `last.pt` |
| Epoch and best fitness | Resume `last.pt` |

This can cause an immediate loss jump and sustained accuracy degradation after resuming.

### Affected condition

The issue occurs when all of the following are true:

1. The original run used `pretrained` as a `str` or `Path`, not the usual Boolean `True`.
2. That value was saved in `last.pt` as part of `train_args`.
3. Training is resumed with `resume=True` or `resume=path/to/last.pt`.

The normal `pretrained=True` case does not enter the problematic `isinstance(..., (str, Path))` branch.

### Minimal reproduction

Start a small run from a YAML architecture while explicitly supplying a pretrained path:

```python
from ultralytics import YOLO

model = YOLO("yolo11n.yaml")
model.train(
    data="coco8.yaml",
    epochs=10,
    pretrained="yolo11n.pt",
)
```

Stop the run after at least one completed epoch so that `last.pt` contains a resumable optimizer state. Then resume it:

```python
from ultralytics import YOLO

model = YOLO("runs/detect/train/weights/last.pt")
model.train(resume=True)
```

The resumed trainer reads `pretrained="yolo11n.pt"` from the saved checkpoint arguments. During
`BaseTrainer.setup_model()`, `load_checkpoint(last.pt)` is called first and `load_checkpoint(yolo11n.pt)` is called
afterward. The latter value becomes the `weights` argument passed to `get_model()`.

The problem can also be verified without relying on metrics by instrumenting or temporarily logging the two
`load_checkpoint()` calls in `BaseTrainer.setup_model()`.

### Relevant code path

Upstream references:

- [`BaseTrainer.setup_model()` on `main`](https://github.com/ultralytics/ultralytics/blob/main/ultralytics/engine/trainer.py)
- [Commit `79b2086bfd` that moved custom `pretrained` loading outside the `.pt` branch](https://github.com/ultralytics/ultralytics/commit/79b2086bfd76e12f2b1bd2aa611ff437fbbcd457)

`BaseTrainer.check_resume()` restores the checkpoint arguments and only reinstates the resume model path:

```python
ckpt_args = load_checkpoint(last)[0].args
self.args = get_cfg(ckpt_args)
self.args.model = self.args.resume = str(last)
```

The saved string-valued `pretrained` argument remains in `self.args`. It is not one of the resume override keys.

`BaseTrainer.setup_model()` then performs both loads:

```python
if str(self.model).endswith(".pt"):
    weights, ckpt = load_checkpoint(self.model)  # resume last.pt
    cfg = weights.yaml
if isinstance(self.args.pretrained, (str, Path)):
    weights, _ = load_checkpoint(self.args.pretrained)  # overwrites resume weights
```

Finally, `resume_training(ckpt)` restores checkpoint state but does not restore the training model weights:

```python
self._load_checkpoint_state(ckpt)  # optimizer, scaler, EMA, best_fitness
```

Passing `pretrained=False` in the resume call is not currently a reliable workaround because `check_resume()` replaces
the current arguments with `ckpt_args`, and `pretrained` is not included in its allowed resume override keys.

### Expected behavior

When `resume` is active, the training model weights and all resumable training state should come from the same resume
checkpoint. A saved custom `pretrained` path should only describe how the original run was initialized; it should not be
loaded again during resume.

### Actual behavior

The training model is rebuilt from the original custom pretrained checkpoint while the optimizer and training progress
are restored from `last.pt`.

### Proposed fix

Do not load `args.pretrained` when resuming:

```diff
-        if isinstance(self.args.pretrained, (str, Path)):
+        if not self.resume and isinstance(self.args.pretrained, (str, Path)):
             weights, _ = load_checkpoint(self.args.pretrained)
```

A regression test could create a resumable checkpoint whose `train_args["pretrained"]` is a different checkpoint path,
then assert that `setup_model()` passes the resume checkpoint model—not the original pretrained model—to `get_model()`.

### Environment observed

- Ultralytics: `8.4.95`
- Repository commit: `d5a1e9472d37b658cada273a86f271c73e57dd54`
- Relevant behavior is also present in the current upstream `main` implementation of `BaseTrainer.setup_model()` at the
  time of reporting.
- OS, Python, and PyTorch versions: please fill in from the reproduction environment.

### Environment

OS: Linux-5.4.0-150-generic-x86_64-with-glibc2.35
Python: 3.11.11 | packaged by conda-forge | (main, Dec  5 2024, 14:17:24) [GCC 13.3.0]
Ultralytics: 8.4.95
PyTorch: 2.6.0+cu118
CUDA: 11.8
cuDNN: 90100
GPU: NVIDIA GeForce RTX 3090

### Minimal Reproducible Example

from ultralytics import YOLO

# Step 1: Start a new training run with `pretrained` specified as a path string.
model = YOLO("yolo11n.yaml")
model.train(
    data="coco8.yaml",
    epochs=10,
    pretrained="yolo11n.pt",
    project="runs/reproduce",
    name="resume_pretrained_bug",
)

### Are you willing to submit a PR?

- [x] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @geiyiren, thank you for the detailed report and reproducible example. The reported resume behavior, where a saved custom `pretrained` path may replace the resume checkpoint’s model weights while other training state is restored, provides sufficient detail for investigation.

An Ultralytics engineer will review the `check_resume()` and `setup_model()` code paths and the proposed regression test soon. This response is automated; thank you also for offering to submit a PR.

**Comment by Y-T-G** [comment]
Thanks for reporting
